Error when building Geant3: cannot find -lROOT::EGPythia6

Hi

I'm trying to build FairSoft (latest in dev branch) in rhel 8.9 but get an error during the building process:

```shell
 cannot find -lROOT::EGPythia6
/opt/rh/gcc-toolset-10/root/usr/bin/ld: cannot find -lROOT::EGPythia6
collect2: error: ld returned 1 exit status
make[5]: *** [CMakeFiles/geant321.dir/build.make:17233: libgeant321.so] Error 1
make[4]: *** [CMakeFiles/Makefile2:110: CMakeFiles/geant321.dir/all] Error 2
make[3]: *** [Makefile:156: all] Error 2
```
I guess this comes from the building of Geant3. It looks like Pythia6 is not built with ROOT, which is required from Geant3.

--- a/cmake/legacy.cmake
+++ b/cmake/legacy.cmake
@@ -349,6 +349,7 @@ ExternalProject_Add(root
     "-Dminuit2=ON"
     "-Dmlp=ON"
     "-Dpyroot=ON"
+    "-Dpythia6=ON"
     "-Dreflex=OFF"
     "-Droofit=ON"
     "-Drpath=ON"

--- a/cmake/legacy.cmake
+++ b/cmake/legacy.cmake
@@ -497,7 +497,7 @@ add_custom_target(source-cache
 
 include(CTest)
 
-foreach(ver IN ITEMS 18.4 18.6 18.8)
+foreach(ver IN ITEMS 18.6 18.8)
   set(TEST_VERSION v${ver}_patches)
   configure_file(test/legacy/fairroot.sh.in ${CMAKE_BINARY_DIR}/test_fairroot_${ver}.sh @ONLY)
   add_test(NAME FairRoot_${ver}

